test/test_classifier.rb

```diff
@@ -46,7 +48,12 @@ def test_instance_classify_nil
   def test_classify_ambiguous_languages
     # Failures are reasonable in some cases, such as when a file is fully valid in more than one language.
     allowed_failures = {
+      # Valid C and C++
       "#{samples_path}/C++/rpc.h" => ["C", "C++"],
+      # Tricky samples
+      "#{samples_path}/C/syscalldefs.h" => ["C", "C++"],
+      "#{samples_path}/C++/Types.h" => ["C", "C++"],
+      "#{samples_path}/R/hello-r.R" => ["R", "Rebol"],
```

Comment:
> and in the case of the last one, absolutely terrible?

Why are we bothering to keep it? There's already a [sample](https://github.com/github/linguist/blob/master/samples/R/import.r) for the `.r` extension, and I doubt the removal of a 4-line _"Hello, world"_ sample is going to impact our classifier much.

Unless the casing of the file-extension matters? (If so, why?)

Reply:
> Why are we bothering to keep it?

I don't think we need to. I didn't know we had this sample until now and I see now it was added over 9 years ago. Lets 🔥 it in another PR.

Reply:
The current classifier overfits. It almost always classifies samples in the training set correctly. This does not necessarily mean it performs better with unseen samples.

Since the new classifier does not exhibit such behavior, that also means that some hard samples are not classified correctly in tests.

Reply:
This is a terrible sample. Replacing it with a real sample solves this and we can remove this. I'll add a better sample and push to this PR.